app.js

```diff
@@ -0,0 +1,83 @@
+const express = require('express');
+const app = express();
+const mongoose = require('mongoose');
+const bp = require('body-parser');
+const rateLimit = require("express-rate-limit");
+const cors = require("cors");
+var cron = require("node-cron");
+
+require('dotenv/config');
+
+app.use(cors());
+app.set("trust proxy", true);
+
+const apiLimiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100,
+});
+
+app.use(bp.urlencoded({ extended: false }));
+app.use(bp.json());
+
+const testRoute = require('./routes/testroute');
+app.use('/test', testRoute);
+
+const scoreRoute = require('./routes/leaderboard');
+app.use('/score', scoreRoute);
+
+const submitRoute = require('./routes/submit');
+app.use('/submit', submitRoute, apiLimiter);
+
+const startingRoute = require("./routes/starting");
+app.use("/starting", startingRoute);
+
+const mapRoute = require("./routes/mapdisplay");
+app.use("/map", mapRoute);
+
+const penaltyRoute = require("./routes/penalty");
+app.use("/penalty", penaltyRoute);
+
+app.get('/', (req, res) => {
+  res.send('homepage');
+});
+
+// to update penalty points every 12 hours
+app.get("/counter", (req, res) => {
+  cron.schedule("0 */12 * * *", () => {
+    player.currentPenaltyPoints = 20;
+  });
+  console.log("penalty");
+});
```

Comment:
Player is undefined. Also, I am not sure why this needs to be inside a route. If you want to reset all player's penalty points to 20 after every 12 hours, just define a cron job which finds all users and sets their points to 20.

```
// utils/resetPenaltyPoints.js

const User = require('../models/User');
const cron = require("node-cron");

cron.schedule("0 */12 * * *", async () => {
    await User.updateMany({}, {
        $set: {
            currentPenaltyPoints: 20,
        },
    });
});
```

```
// app.js

...
require('./utils/resetPenaltyPoints');
...
```